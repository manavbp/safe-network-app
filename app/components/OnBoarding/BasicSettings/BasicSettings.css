.Base {
    justify-content: center;
    align-items: center;
    flex-direction: column;
    padding: 0 24px;
}

.Title {
    padding-top: 10%;
    margin-bottom: 16px;
}
