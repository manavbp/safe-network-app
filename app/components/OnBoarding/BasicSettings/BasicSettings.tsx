import React, { Component } from 'react';
import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import Toolbar from '@material-ui/core/Toolbar';
import Box from '@material-ui/core/Box';
import Typography from '@material-ui/core/Typography';
import styles from './BasicSettings.css';

import { UserPreferences } from '$Definitions/application.d';
import { Preferences } from '$Components/Preferences';

interface Props {
    userPreferences: UserPreferences;
    setUserPreferences: Function;
    triggerSetAsTrayWindow: Function;
    autoLaunch: Function;
    isTrayWindow: boolean;
}

export const BasicSettings = ( properties ) => {
    const {
        userPreferences,
        setUserPreferences,
        triggerSetAsTrayWindow,
        autoLaunch,
        isTrayWindow
    } = properties;
    const requiredItems = {
        autoUpdate: true,
        pinToMenuBar: true,
        launchOnStart: true,
        showDeveloperApps: true
    };
    return (
        <Paper
            className={styles.Base}
            elevation={0}
            aria-label="BasicSettingsPage"
        >
            <Box>
                <Typography className={styles.Title} variant="h5">
                    Basic Settings
                </Typography>
                <Typography>
                    Choose some basic settings. You can always change these
                    later.
                </Typography>
                <Grid item xs={12}>
                    <Preferences
                        isTrayWindow={isTrayWindow}
                        userPreferences={userPreferences}
                        requiredItems={requiredItems}
                        onChange={setUserPreferences}
                        onChangeLaunchOnStart={autoLaunch}
                        onChangePinToMenu={triggerSetAsTrayWindow}
                    />
                </Grid>
            </Box>
        </Paper>
    );
};
