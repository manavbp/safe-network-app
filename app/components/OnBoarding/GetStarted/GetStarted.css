.Base {
    width: 100%;
    height: 100%;
    min-height: 600px;
    background: linear-gradient(156.98deg, #fa541c -164.31%, #454545 101.04%);
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
}

.LogoBase {
    display: flex;
    width: 80px;
    height: 80px;
    background-color: #fa541c;
    border-radius: 50%;
    justify-content: center;
    box-shadow: 0 0 100px #fa541c, 0 0 50px #fa541c,
        0 0 23px rgba(0, 0, 0, 0.25);
}

.Container {
    text-align: center;
    margin-top: 60px;
    color: #fff;
    padding: 0 40px;
}

.Title {
    margin-bottom: 20px;
}

.GetStartedButton {
    text-transform: uppercase;
    height: 36px;
    margin-top: 30px;
    background-color: #ff5722;
    color: #fff;
}
