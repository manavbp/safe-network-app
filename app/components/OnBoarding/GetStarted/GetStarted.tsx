import React from 'react';
import Typography from '@material-ui/core/Typography';
import Paper from '@material-ui/core/Paper';
import Box from '@material-ui/core/Box';
import Fab from '@material-ui/core/Fab';
// @ts-ignore
import { INTRO } from '$App/constants/routes.json';

// @ts-ignore
import Logo from '$App/assets/images/logo.svg';
import styles from './GetStarted.css';

interface Props {
    onClickGetStarted: Function;
    history?: History;
}

export const GetStarted = ( properties ) => {
    const { onClickGetStarted, history } = properties;
    return (
        <Paper className={styles.Base} aria-label="GetStarted">
            <Box className={styles.LogoBase}>
                <img src={Logo} alt="Launchpad logo" />
            </Box>
            <Box className={styles.Container}>
                <Typography className={styles.Title} variant="h5">
                    SAFE Launchpad
                </Typography>
                <Typography>
                    All the apps you need to try the SAFE Network
                </Typography>
                <Fab
                    className={styles.GetStartedButton}
                    variant="extended"
                    size="medium"
                    aria-label="GetStarted"
                    onClick={() => {
                        onClickGetStarted();
                        // @ts-ignore
                        history.push( INTRO );
                    }}
                >
                    Get Started
                </Fab>
            </Box>
        </Paper>
    );
};
