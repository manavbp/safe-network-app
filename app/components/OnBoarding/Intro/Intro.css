.Base {
    width: 100%;
    height: 100%;
    background-image: url('../../../assets/images/on_boarding_popup_bg.png');
    background-repeat: no-repeat;
    background-size: contain;
    display: 'flex';
    justify-content: center;
    align-items: center;
    flex-direction: column;
}

.Box {
    padding: '0 24px';
}

.Title {
    padding-top: 450px;
    text-align: center;
    margin-bottom: 16px;
}

.Description {
    text-align: center;
}
