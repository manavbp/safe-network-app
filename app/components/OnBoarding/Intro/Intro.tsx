import React from 'react';
import Typography from '@material-ui/core/Typography';
import Paper from '@material-ui/core/Paper';
import Box from '@material-ui/core/Box';
import styles from './Intro.css';

export const Intro = () => {
    return (
        <Paper className={styles.Base} elevation={0} aria-label="IntroPage">
            <Box className={styles.Container}>
                <Typography className={styles.Title} variant="h5">
                    One Place for All SAFE Apps
                </Typography>
                <Typography className={styles.Description}>
                    A one-stop shop to access all SAFE Apps and manage instant
                    app updates.
                </Typography>
            </Box>
        </Paper>
    );
};
