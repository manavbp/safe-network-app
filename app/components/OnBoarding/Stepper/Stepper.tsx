import React from 'react';
import { withStyles, styled } from '@material-ui/core/styles';
import indigo from '@material-ui/core/colors/indigo';
import Button from '@material-ui/core/Button';
import MobileStepper from '@material-ui/core/MobileStepper';
import {
    ON_BOARDING,
    BASIC_SETTINGS,
    INTRO,
    HOME
} from '$Constants/routes.json';

interface Props {
    showButtons?: boolean;
    noElevation?: boolean;
    theme?: 'default' | 'white';
    onNext: Function;
    onBack: Function;
    steps: number;
    activeStep: number;
    history?: History;
}

const defaultTheme = {
    root: {
        backgroundColor: '#fff'
    },
    dot: {
        backgroundColor: indigo[100]
    },
    dotActive: {
        backgroundColor: indigo[800]
    }
};

const whiteTheme = {
    root: {
        backgroundColor: 'transparent'
    },
    dot: {
        backgroundColor: 'rgba(#fff, 0.3)'
    },
    dotActive: {
        backgroundColor: '#fff'
    }
};

export const Stepper = ( properties ) => {
    const {
        showButtons,
        theme,
        steps,
        activeStep,
        onBack,
        onNext,
        noElevation,
        history
    } = properties;

    const NavButton = styled( Button )( {
        color: indigo[800],
        visibility: showButtons ? 'visible' : 'hidden'
    } );

    const NextButton = (
        <NavButton
            size="small"
            aria-label="OnBoardingNextButton"
            onClick={() => {
                onNext();
                // @ts-ignore
                history.push( BASIC_SETTINGS );
            }}
        >
            Next
        </NavButton>
    );
    const BackButton = (
        <NavButton
            size="small"
            aria-label="OnBoardingBackButton"
            onClick={() => {
                onBack();
                // @ts-ignore
                activeStep === 1
                    ? history.push( ON_BOARDING )
                    : history.push( INTRO );
            }}
        >
            Back
        </NavButton>
    );

    const StyledStepper = withStyles(
        theme === 'white' ? whiteTheme : defaultTheme
    )( MobileStepper );

    return (
        <StyledStepper
            elevation={noElevation ? 0 : 4}
            variant="dots"
            steps={steps}
            position="bottom"
            activeStep={activeStep}
            nextButton={NextButton}
            backButton={BackButton}
        />
    );
};

Stepper.defaultProps = {
    showButtons: true,
    steps: 3,
    theme: 'default',
    activeStep: 0,
    noElevation: false
};
