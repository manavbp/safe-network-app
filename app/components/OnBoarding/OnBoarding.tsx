import React, { Component } from 'react';
import { History } from 'history';
import Grid from '@material-ui/core/Grid';
import { Switch, Route } from 'react-router';
import { styled } from '@material-ui/core/styles';
import {
    UserPreferences,
    AppPreferences,
    Preferences,
    LaunchpadState
} from '$Definitions/application.d';

import { Stepper } from './Stepper/Stepper';
import { GetStarted } from './GetStarted';
import { Intro } from './Intro';
import { BasicSettings } from './BasicSettings/BasicSettings';
import {
    HOME,
    INTRO,
    BASIC_SETTINGS,
    ON_BOARDING
} from '$Constants/routes.json';

interface Props {
    setAppPreferences: Function;
    aliasShouldOnboard: Function;
    triggerSetAsTrayWindow: Function;
    userPreferences: UserPreferences;
    appPreferences: AppPreferences;
    setUserPreferences: Function;
    getUserPreferences: Function;
    storePreferences: Function;
    isTrayWindow: boolean;
    setOnboardCompleted: Function;
    autoLaunch: Function;
    history?: History;
}

const Base = styled( Grid )( {
    position: 'absolute',
    top: 0,
    left: 0,
    width: '100%',
    height: '100%'
} );

export class OnBoarding extends React.Component<Props> {
    state = {
        currentPosition: 0
    };

    totalSteps = 3;

    componentWillMount() {
        const { appPreferences, history, getUserPreferences } = this.props;

        if ( !appPreferences.shouldOnboard ) {
            history.push( HOME );
        }
        getUserPreferences();
    }

    completed = () => {
        const {
            storePreferences,
            userPreferences,
            autoLaunch,
            setAppPreferences
        } = this.props;

        const appPreferences: AppPreferences = {
            shouldOnboard: false
        };

        const newUserPreferences: UserPreferences = {
            ...userPreferences
        };

        const preferences: Preferences = {
            userPreferences: { ...newUserPreferences },
            appPreferences
        };

        storePreferences( preferences );
        setAppPreferences( { shouldOnboard: false } );

        if ( userPreferences.launchOnStart ) {
            autoLaunch( true );
        }
    };

    onNext = () => {
        const { currentPosition } = this.state;
        const position = currentPosition;

        if ( currentPosition < this.totalSteps - 1 ) {
            this.setState( {
                currentPosition: currentPosition + 1
            } );
        } else {
            this.completed();
        }
    };

    onBack = () => {
        const { currentPosition } = this.state;
        const position = currentPosition;

        if ( currentPosition > 0 ) {
            this.setState( {
                currentPosition: currentPosition - 1
            } );
        }
    };

    render() {
        const { currentPosition } = this.state;
        const {
            userPreferences,
            setUserPreferences,
            getUserPreferences,
            storePreferences,
            isTrayWindow,
            triggerSetAsTrayWindow,
            autoLaunch,
            history
        } = this.props;

        const isGetStarted = currentPosition === 0;

        return (
            <Base container>
                <Grid item xs={12}>
                    <Switch>
                        <Route
                            exact
                            path={ON_BOARDING}
                            component={() => (
                                <GetStarted
                                    onClickGetStarted={this.onNext}
                                    // @ts-ignore
                                    history={history}
                                />
                            )}
                        />
                        <Route path={INTRO} component={() => <Intro />} />
                        <Route
                            path={BASIC_SETTINGS}
                            component={() => (
                                <BasicSettings
                                    userPreferences={userPreferences}
                                    setUserPreferences={setUserPreferences}
                                    triggerSetAsTrayWindow={
                                        triggerSetAsTrayWindow
                                    }
                                    autoLaunch={autoLaunch}
                                    isTrayWindow={isTrayWindow}
                                />
                            )}
                        />
                    </Switch>
                </Grid>
                <Stepper
                    theme={isGetStarted ? 'white' : 'default'}
                    showButtons={!isGetStarted}
                    noElevation={isGetStarted}
                    onNext={this.onNext}
                    onBack={this.onBack}
                    steps={this.totalSteps}
                    activeStep={this.state.currentPosition}
                    // @ts-ignore
                    history={history}
                />
            </Base>
        );
    }
}
