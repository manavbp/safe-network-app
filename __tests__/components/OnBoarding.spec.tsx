import React from 'react';
import Fab from '@material-ui/core/Fab';
import { createShallow } from '@material-ui/core/test-utils';

import { OnBoarding } from '$Components/OnBoarding/OnBoarding';
import { GetStarted } from '$Components/OnBoarding/GetStarted';
import { Intro } from '$Components/OnBoarding/Intro/Intro';
import { BasicSettings } from '$Components/OnBoarding/BasicSettings/BasicSettings';
import { Stepper } from '$Components/OnBoarding/Stepper/Stepper';

const shallow = createShallow();

const shallowSetup = ( propOverrides? ) => {
    const props = Object.assign(
        {
            appPreferences: {
                shouldOnboard: true
            },
            userPreferences: {
                autoUpdate: false,
                pinToMenuBar: false,
                launchOnStart: false,
                showDeveloperApps: false,
                warnOnAccessingClearnet: false
            },
            getUserPreferences: jest.fn(),
            setUserPreferences: jest.fn(),
            setAppPreferences: jest.fn(),
            pinToTray: jest.fn(),
            autoLaunch: jest.fn(),
            storePreferences: jest.fn(),
            setOnboardCompleted: jest.fn(),
            history: {
                push: jest.fn()
            }
        },
        propOverrides
    );

    const wrapper = shallow( <OnBoarding {...props} /> );

    return {
        props,
        wrapper
    };
};

describe( 'On Boarding', () => {
    it( 'render', () => {
        const { wrapper } = shallowSetup();
        expect( wrapper ).toMatchSnapshot();
    } );

    it( 'stepper should show current position', () => {
        const { wrapper } = shallowSetup();
        const instance = wrapper.instance();
        expect( wrapper.find( Stepper ).props().steps ).toEqual(
            instance.totalSteps
        );

        wrapper.setState( { currentPosition: 0 } );
        expect( wrapper.find( Stepper ).props().activeStep ).toEqual( 0 );
        wrapper.setState( { currentPosition: 1 } );
        expect( wrapper.find( Stepper ).props().activeStep ).toEqual( 1 );
        wrapper.setState( { currentPosition: 2 } );
        expect( wrapper.find( Stepper ).props().activeStep ).toEqual( 2 );
    } );

    it( 'Go to Basic settings page on clicking next', () => {
        const { wrapper } = shallowSetup();
        wrapper.setState( { currentPosition: 1 } );
        wrapper
            .find( Stepper )
            .props()
            .onNext();
        expect( wrapper.state().currentPosition ).toEqual( 2 );
    } );

    it( 'Go back to Intro page from Basic settings page on clicking back', () => {
        const { wrapper } = shallowSetup();
        wrapper.setState( { currentPosition: 2 } );
        wrapper
            .find( Stepper )
            .props()
            .onBack();
        expect( wrapper.state().currentPosition ).toEqual( 1 );
    } );

    it( 'Go back to Get Started page from Intro page on clicking back', () => {
        const { wrapper } = shallowSetup();
        wrapper.setState( { currentPosition: 1 } );
        wrapper
            .find( Stepper )
            .props()
            .onBack();
        expect( wrapper.state().currentPosition ).toEqual( 0 );
    } );

    it( 'complete on clicking Next from basic settings page', () => {
        const { wrapper, props } = shallowSetup();
        wrapper.setState( { currentPosition: 2 } );
        wrapper
            .find( Stepper )
            .props()
            .onNext();
        expect( props.setAppPreferences ).toHaveBeenCalled();
    } );
} );
