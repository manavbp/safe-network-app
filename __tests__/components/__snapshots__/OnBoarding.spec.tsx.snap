// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`On Boarding render 1`] = `ShallowWrapper {}`;
